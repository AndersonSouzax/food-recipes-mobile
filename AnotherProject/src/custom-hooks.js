import { useState, useEffect } from 'react';
import AsyncStorage from '@react-native-community/async-storage';

export const useAsyncStorage = () => {

	const [user, setUser] = useState(null);

	useEffect(() => {

		const seach = async () => {

			try {

		    const value = await AsyncStorage.getItem('FoodRecipeToken');

		    if(value !== null) {
		    	const token = JSON.parse(value);
		      setUser(token);
		    }else{
		    	setUser('error');
		    }
		  } catch(e) {
		    setUser('error');
		  }
		}

		seach();

	}, []);

	return user;

}