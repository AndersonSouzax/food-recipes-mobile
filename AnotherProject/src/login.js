import React, { useState } from 'react';

import { 
	StyleSheet, 
	TextInput, 
	ScrollView, 
	View,
	Image, 
	Text, 
	StatusBar,
	Pressable 
} from 'react-native';

import AsyncStorage from '@react-native-community/async-storage';
import { CommonActions } from '@react-navigation/native';

import API from './services/api';

export default function Login({ message, navigation }){

	const [username, setUsername] = useState('');
	const [password, setPassword] = useState('');
	const [error, setError] = useState(message ? message : '');

	const handleSignInPress = async () => {

		if(password.length < 6){ 

			setError('The password needs to have at least 6 chars'); 

		}else{

			try{

				const response = await API.login({ username, password });

				await AsyncStorage.setItem('FoodRecipeToken', response.data.token);

		    const resetAction = CommonActions.reset({
		      index: 0,
		      routes: [
		        { name: 'Main' },
		      ],
		    });

    		navigation.dispatch(resetAction);

			}catch(e){
				setError('Login error, check your credentials and try again..');
			}	
		}
	};

	const handleUsername = (username) => {
		setUsername(username);
	};

	const handlePassword = (password) => {
		setPassword(password);
	};

	const handleKeyPress = () => {

		if(error.length !== 0){
			setError('');
		}

	};

	return (

		<View style={styles.container}>

			<StatusBar hidden />

			<Image style={styles.logo} source={require('./images/tomato.png')} 
				resizeMode="contain" />

			<Text style={styles.title}> Food Recipes </Text>

			<TextInput onKeyPress={handleKeyPress} onChangeText={handleUsername} 
				value={username} style={styles.input} placeholder="email" />
			
			<TextInput onKeyPress={handleKeyPress} onChangeText={handlePassword} 
				value={password} style={styles.input} placeholder="password"/>

			{ error.length !== 0 && <Text testID="error"> { error } </Text> }

			<Pressable onPress={handleSignInPress} disabled={!username && !password} 
				testID="loginButton" style={styles.button}>
				
				<Text style={styles.buttonTxt}>Login</Text>

			</Pressable>

		</View>

	);
}

// Style

const styles = StyleSheet.create({

	container : {
		flex: 1,
	  alignItems: 'center',
	  justifyContent: 'center',
	  backgroundColor: '#FFFFFF',
	},
	button : {
		padding: 20,
	  borderRadius: 5,
	  alignSelf: 'stretch',
	  margin: 15,
	  marginHorizontal: 20,
		backgroundColor: '#EA1D2C',
	},
	buttonTxt :{
		color: '#FFF',
	  fontWeight: 'bold',
	  fontSize: 16,
	  textAlign: 'center',
	},
	logo : {
		height: '30%',
  	marginBottom: 20,
	},
	title : {
		fontFamily: 'cursive',
		fontWeight: 'bold',
		fontSize: 32,
		marginBottom: 25,
		color: '#404040'
	},
	input : {
		paddingHorizontal: 20,
	  paddingVertical: 15,
	  borderRadius: 5,
	  backgroundColor: '#FFFFFF',
	  alignSelf: 'stretch',
	  marginBottom: 15,
	  marginHorizontal: 20,
	  fontSize: 16,
	  borderColor: '#EA1D2C',
	  borderWidth: 1,
	}
});


