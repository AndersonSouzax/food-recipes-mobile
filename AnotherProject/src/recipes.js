import React, { useState, useEffect, useContext } from 'react';
import AsyncStorage from '@react-native-community/async-storage';
import { CommonActions } from '@react-navigation/native';

import { 
	StyleSheet, 
	FlatList, 
	View,
	Image,
	Text, 
	StatusBar,
	Pressable,
} from 'react-native';

import { 
	title, 
	container, 
	button, 
	buttonTxt,
} from './styles';

import { Appbar, Menu, Card, Title, Paragraph } from 'react-native-paper';

import API from './services/api';
import { AuthContext } from './authcontext';

export default function Recipes({ route, navigation }){

	const { user } = route.params;

	const [recipes, setRecipes] = useState(null);
	const [loading, setLoading] = useState({ loading: false, error: '', obj : '' });
	const [reload, setReload] = useState({ type : 'ALL' });
	const [stored, setStored] = useState(user);
	const [menuVisible, setMenuVisible] = useState(false);
	const context = useContext(AuthContext);

	// Function related to the same hook: recipes
	useEffect(() => {

		let mounted = true;

		const fetchAll = async () => { 

			if(loading.loading){ return; }

			try{

				setLoading({ error : '', loading : true, 
					'obj' : reload.type === 'ALL' ? 'All The Recipes' 
						: 'Only The Recipes Created By you'});

				let path = reload.type === 'ALL' 
					? '/recipe' 
					: `/recipe?user=${stored.id}`;

				const response = await API.request(path, 'get', stored.token, null);

				if(mounted){

					setLoading({ loading : false, obj : '', error: '' });

					setRecipes(response.data);

				}

			}catch(e){

				setLoading({ loading : false, obj : '', 
					error: `An error has occurred while getting recipes: ${e.message}` });

			}
		}

		fetchAll();

		//Avoid make changes on unmounted component...
		return () => mounted = false;

	}, [reload, stored]);

	const handleMy = () => { setReload({ type : 'MY_RECIPES' }); }

	const handleAll = () => { setReload({ type : 'ALL' }); }

	const handleLogout = async () => {

		try{

			await AsyncStorage.removeItem('FoodRecipeToken');

			context.signOut();

		}catch(e){
			setLoading({ loading: false, obj: '', 
				error: 'An error occurred while removing user token and logging out'});
		}
	}

	const navigateToSingle = (item) => {
  	navigation.navigate('SingleRecipe', {
  		recipe : item,
  		user : stored,
  	});
  }

  const openMenu = () => setMenuVisible(true);

  const closeMenu = () => setMenuVisible(false);

	const recipeRender = ({ item }) => (
  	<Card accessibilityLabel="A Recipe" 
  		style={styles.cardStyle}
  		onPress={ () => { navigateToSingle(item); } }>
    	<Card.Cover source={ item.category.image ? 
    		{ uri: item.category.image } : require('./images/default_recipe.png') } />
    	<Card.Content>
      	<Title>{item.title || 'Recipe Title' }</Title>
      	<Paragraph>{ item.description || 'Delicious food...'}</Paragraph>
    	</Card.Content>
    	
  	</Card>
	);

	return (
		<>

		<StatusBar backgroundColor="#EA1D2C" />

    <Appbar.Header style={styles.appBar}>

    	<Appbar.Content title="Food Recipes" style={styles.title}/>

    	<Menu
			  onDismiss={closeMenu}
			  visible={menuVisible}
			  style={styles.menu}
			  anchor={
			     <Appbar.Action
			     	testID="user-menu"
			      disabled={loading.loading}
			      color="white"
			      icon="face"
			      onPress={openMenu}
			     />
			   }>
			  <Menu.Item testID="logout-prs" icon="undo" 
			  	title="Logout" onPress={handleLogout} 
			  	style={styles.menuItem}/>

			  <Menu.Item testID="logout-prs" icon="fruit-watermelon"
			  	title="Add Recipe" onPress={ () => { navigateToSingle(); } } 
			  	style={styles.menuItem}/>

			</Menu>

    </Appbar.Header>

			<View style={styles.listView}>

				{ 
					recipes && ( 
						recipes.length > 0 ?
							<FlatList testID="recipes-list" data={recipes} 
								renderItem={recipeRender} 
								keyExtractor={ () => '_' + Math.random()
								.toString(36)
								.substr(2, 9) 
								}
								style={styles.list}/>
						: 
							<Text> No recipes received for the token...</Text>
					)
				}

				{ loading.loading ? <Text testID="load-txt"> Loading {loading.obj}...</Text> : <></> }
				{ loading.error ? <Text testID="error"> { loading.error } </Text> : <></> }
			</View>

			<Appbar style={styles.bottom}>

				<Appbar.Action testID="all-recipes" icon="filmstrip" 
					onPress={handleAll} disabled={loading.loading} />
		    <Appbar.Action testID="my-recipes" icon="emoticon-cool-outline" 
		    	onPress={handleMy} disabled={loading.loading} />

		  </Appbar>
	  </>
	);
};

const styles = StyleSheet.create({
  bottom: {
  	backgroundColor: '#EA1D2C',
    position: 'absolute',
    left: 0,
    right: 0,
    bottom: 0,
    flex: 1,
    justifyContent : 'space-around'
  },
  appBar: {
  	backgroundColor: '#EA1D2C',
  },
  title : {
  	fontFamily: 'cursive',
		fontWeight: 'bold',
		alignItems: 'center'
  },
  cardStyle: {
  	marginBottom: 7,
  },
  list : {
  	width : '95%',
  },
  listView: { 
  	alignItems: 'center', 
  	justifyContent: 'center',
  	flexGrow: 0,
  	height: '77%'
  },
  menu : {
    marginTop: '14%',
  },
  menuItem : {
  	color : '#EA1D2C',
  }
});

