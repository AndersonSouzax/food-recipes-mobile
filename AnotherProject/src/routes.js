import 'react-native-gesture-handler';
import React, { useState, useEffect } from 'react';
import { View, Text } from 'react-native';
import { NavigationContainer } from '@react-navigation/native';
import { createStackNavigator } from '@react-navigation/stack';
import AsyncStorage from '@react-native-community/async-storage';

import Login from './login';
// import SignUp from './pages/signUp';
// import Main from './pages/main';

function Main(){
	return <> <p>Main</p> </>;
}

const Stack = createStackNavigator();

export default function Routes () {

	const [user, setUser] = useState({ loading : true, stored : null, error : '' });

	useEffect(() => {

		const seach = async () => {

			try {

	    	const value = await AsyncStorage.getItem('FoodRecipeToken');

		    if(value !== null) {
		    	const token = JSON.parse(value);
		      setUser({ ...user, loading : false, stored : token });
		    }else{
		    	setUser({ ...user, loading : false });
		    }
		  } catch(e) {
		    setUser({...user, loading : false, error : 'Exception on AsyncStorage'});
		  }
		}

	seach();

	}, []);

	if(user.loading){
		return ( 
			<View style={{ flex: 1, alignItems: 'center', justifyContent: 'center' }}>
				<Text>Loading...</Text>
			</View>
		);
	}

	return ( 
		<NavigationContainer>
		  <Stack.Navigator>
		    { user.stored === null ? ( 
					<Stack.Screen name="Login" component={Login} 
						message={ user.error ? user.error : null }/>
	 				) : (
	 					<Stack.Screen name="Main" component={Main}/>
	 				) } 
		  </Stack.Navigator>
		</NavigationContainer>
	);
 };