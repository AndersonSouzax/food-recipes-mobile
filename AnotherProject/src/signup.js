import React, { useState } from 'react';

import { 
	StyleSheet, 
	TextInput, 
	ScrollView, 
	View,
	Image,
	ImageBackground, 
	Text, 
	StatusBar,
	Pressable 
} from 'react-native';

import { 
	title, 
	container, 
	button, 
	buttonTxt, 
	linkButton, 
	linkButtonTxt, 
	input
} from './styles';

export default function SignUp({ navigation }){

	const [username, setUsername] = useState('');
	const [password, setPassword] = useState('');
	const [repeat, setRepeat] = useState('');
	const [error, setError] = useState('');

	const handleKeyPress = () => {};

	const handleUsername = () => {};

	const handlePassword = () => {};

	const handlePasswordAGain = () => {};

	const handleSignUpPress = () => {};

	return (

		<View style={styles.container}>

			<StatusBar hidden />

			<ImageBackground source={require('./images/vegetables.jpg')} style={styles.backImage}>

				<Text style={styles.title}> Food Recipes </Text>

				<Text style={styles.subTitle}> 
					Sign up and start discovering a world of delicious foods to cook and eat with family
				</Text>

				<TextInput onKeyPress={handleKeyPress} onChangeText={handleUsername} 
					value={username} style={styles.input} placeholder="email" />
				
				<TextInput onKeyPress={handleKeyPress} onChangeText={handlePassword} 
					value={password} style={styles.input} placeholder="password"/>

				<TextInput onKeyPress={handleKeyPress} onChangeText={handlePasswordAGain} 
					value={repeat} style={styles.input} placeholder="repeat password"/>

				{ error.length !== 0 && <Text testID="error"> { error } </Text> }

				<Pressable onPress={handleSignUpPress} disabled={!username && !password && !repeat} 
					testID="signUpButton" style={styles.button}>
					
					<Text style={styles.buttonTxt}>Sign Up</Text>

				</Pressable>

	    </ImageBackground>

		</View>
	);
};

const styles = StyleSheet.create({
	container: container,
	title: title,
	input: input,
	button: button,
	buttonTxt: buttonTxt,
	subTitle: {
		fontFamily: 'cursive',
		fontSize: 22,
		marginBottom: 25,
		textAlign: 'center',
		color : 'blue'
	},
	backImage : {
		flex: 1,
		alignItems: 'center',
    resizeMode: "cover",
    justifyContent: "center",
	}
});