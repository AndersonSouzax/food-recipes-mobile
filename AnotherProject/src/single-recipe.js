import React from 'react';

import { 
	StyleSheet,
	View,
	Image,
	Text, 
	StatusBar,
	Pressable,
} from 'react-native';

export default function SingleRecipe({ navigation }){

	return (<Text>reeeeeeeeeeee</Text>);
}