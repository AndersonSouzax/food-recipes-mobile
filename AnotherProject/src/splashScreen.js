import React from 'react';

import tomato from './images/tomato.png';

export default function SplashScreen(){

	
};

