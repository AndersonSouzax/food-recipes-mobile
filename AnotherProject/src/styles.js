
const	title = {
	fontFamily: 'cursive',
	fontWeight: 'bold',
	fontSize: 32,
	marginBottom: 25,
	color: '#404040'
};

const container = {
	flex: 1,
  alignItems: 'center',
  justifyContent: 'center',
  backgroundColor: '#FFFFFF',
};

const	button = {
	padding: 20,
  borderRadius: 5,
  alignSelf: 'stretch',
  margin: 15,
  marginHorizontal: 10,
	backgroundColor: '#EA1D2C',
};

const	linkButton = {
	padding: 5,
  alignSelf: 'stretch',
  marginTop: 7,
  marginHorizontal: 20,
};

const	linkButtonTxt = {
	color: '#EA1D2C',
  fontSize: 16,
  textAlign: 'center',
  textDecorationLine: 'underline',
};

const	buttonTxt = {
		color: '#FFF',
	  fontWeight: 'bold',
	  fontSize: 16,
	  textAlign: 'center',
};

const	input = {
	paddingHorizontal: 20,
  paddingVertical: 15,
  borderRadius: 5,
  backgroundColor: '#FFFFFF',
  alignSelf: 'stretch',
  marginBottom: 15,
  marginHorizontal: 20,
  fontSize: 16,
  borderColor: '#EA1D2C',
  borderWidth: 1,
};

export { title, container, button, buttonTxt, linkButton, linkButtonTxt, input };