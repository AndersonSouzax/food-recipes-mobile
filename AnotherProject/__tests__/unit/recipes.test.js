import 'react-native';
import React from 'react';
import { render, waitFor, act, fireEvent } from '@testing-library/react-native';

import { AuthContext } from '../../src/authcontext';
import Recipes from '../../src/recipes';

import { Provider as PaperProvider } from 'react-native-paper';

jest.mock('../../src/services/api');

jest.setTimeout(20000);

const mockRoute = { 
	params: { user : { token: '9340349njfnidfbuierf', name : 'And', id : 2 } }
};

test('renders correctly', () => {
  const component = render(<Recipes route={mockRoute}/>).toJSON();
  expect(component).toMatchSnapshot();
});

// Almost the same code for all and my recipes request
describe('Fetching recipes', () => {

	test('Display all the recipes after loading', async () => {

		const { getByTestId } = render(
			<Recipes navigation={null} route={mockRoute} />
		);

		await waitFor(() => getByTestId('recipes-list'), { timeout: 19000 });

	});

	test('Load only user\'s created recipes', async () => {

			const component = render(<Recipes navigation={null} route={mockRoute} />);
			
			await act(async () => {

				const { getByTestId } = component;

				await waitFor(() => getByTestId('recipes-list'));

				const button = getByTestId('my-recipes');

				fireEvent.press(button);

				await (async () => {

					// The button must be disabled during loading...
					expect(button.props.disabled).toBe(true);

					await waitFor(() => getByTestId('load-txt'));

					await waitForElementToBeRemoved(() => getByTestId('load-txt'));

					const after = component.toJSON();

					expect(after).toMatchSnapshot();

				});

			});

	});

	test('Load all the recipes', async () => {

			const component = render(<Recipes navigation={null} route={mockRoute} />);
			
			await act(async () => {

				const { getByTestId } = component;

				await waitFor(() => getByTestId('recipes-list'));

				const button = getByTestId('all-recipes');

				fireEvent.press(button);

				await (async () => {

					// The button must be disabled during loading...
					expect(button.props.disabled).toBe(true);

					await waitFor(() => getByTestId('load-txt'));

					await waitForElementToBeRemoved(() => getByTestId('load-txt'));

					const after = component.toJSON();

					expect(after).toMatchSnapshot();

				});
			
			});

	});

// 	test('Notify error on data fetching', async () => { 
// 
// 		const { result, waitForNextUpdate } = renderHook(() => Recipes());
// 
// 		await waitForNextUpdate();
// 
// 		expect(result.current.loading.error).not.toHaveLength(0);
// 
// 	});
	

});

test('User log out', async () => {

	const navigationMock = { dispatch: jest.fn() };

	const mockContext = { signOut: jest.fn() }

	const component = render(
		
			<AuthContext.Provider value={mockContext}>
				<Recipes navigation={navigationMock} route={mockRoute} />
			</AuthContext.Provider>
		
	);

	await act(async () => {

		const { getByTestId } = component;

		await waitFor(() => getByTestId('recipes-list'));

		fireEvent.press(getByTestId('user-menu'));

		await (async () => {

			const button = getByTestId('logout-prs');

			await fireEvent.press(button);

			expect(mockContext.signOut.mock.calls.length).toBe(1);

		});

	});

});

test('Navigate to Single Recipe for Creation/Editing', async () => {

	const navigationMock = { navigate: jest.fn() };

	const component = render(
		<Recipes navigation={navigationMock} route={mockRoute} />
	);

	await act(async () => {

		const { getByTestId, getAllByAccessibilityLabel } = component;

		await waitFor(() => getByTestId('recipes-list'));

		const recipesList = component.getAllByAccessibilityLabel('A Recipe');

		const recipe = recipesList[0];

		await fireEvent.press(recipe);

		expect(navigationMock.navigate.mock.calls.length).toBe(1);

		expect(navigationMock.navigate.mock.calls[0][1]).not.toBeNull();

	});
});

