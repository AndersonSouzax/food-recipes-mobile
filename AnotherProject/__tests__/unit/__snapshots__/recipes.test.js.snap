// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Array [
        Object {
          "backgroundColor": "#EA1D2C",
          "elevation": 4,
          "zIndex": 0,
        },
        Object {
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 3,
            "width": 0,
          },
          "shadowOpacity": 0.24,
          "shadowRadius": 4,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#EA1D2C",
          "elevation": 0,
          "flexDirection": "row",
          "height": 56,
          "marginTop": 0,
          "paddingHorizontal": 4,
        }
      }
    >
      <View
        style={
          Object {
            "width": 48,
          }
        }
      />
      <View
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "flex": 1,
              "paddingHorizontal": 12,
            },
            Array [
              false,
              Object {
                "alignItems": "center",
              },
              Object {
                "fontFamily": "cursive",
                "fontWeight": "bold",
              },
            ],
          ]
        }
      >
        <Text
          accessibilityRole="header"
          accessibilityTraits="header"
          accessible={true}
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
                "textAlign": "left",
              },
              Array [
                Object {
                  "color": "#ffffff",
                  "fontFamily": "System",
                  "fontWeight": "400",
                },
                Object {
                  "fontSize": 17,
                },
                undefined,
              ],
            ]
          }
        >
          Food Recipes
        </Text>
      </View>
      <View
        collapsable={false}
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "disabled": false,
            }
          }
          accessible={true}
          focusable={true}
          hitSlop={
            Object {
              "bottom": 6,
              "left": 6,
              "right": 6,
              "top": 6,
            }
          }
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Array [
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "margin": 6,
                  "overflow": "hidden",
                },
                Object {
                  "borderRadius": 18,
                  "height": 36,
                  "width": 36,
                },
                false,
                undefined,
              ],
            ]
          }
          testID="user-menu"
        >
          <View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "justifyContent": "center",
                  },
                  Object {
                    "height": 24,
                    "width": 24,
                  },
                ]
              }
            >
              <View
                style={
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "opacity": 1,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "transform": Array [
                      Object {
                        "rotate": "0deg",
                      },
                    ],
                  }
                }
              >
                <Text
                  accessibilityElementsHidden={true}
                  allowFontScaling={false}
                  importantForAccessibility="no-hide-descendants"
                  pointerEvents="none"
                  style={
                    Array [
                      Object {
                        "color": "white",
                        "fontSize": 24,
                      },
                      Array [
                        Object {
                          "transform": Array [
                            Object {
                              "scaleX": 1,
                            },
                          ],
                        },
                        Object {
                          "backgroundColor": "transparent",
                        },
                      ],
                      Object {
                        "fontFamily": "Material Design Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  󰙃
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTSafeAreaView>,
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  />,
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#EA1D2C",
        "bottom": 0,
        "elevation": 4,
        "flexDirection": "row",
        "height": 56,
        "left": 0,
        "paddingHorizontal": 4,
        "position": "absolute",
        "right": 0,
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 3,
          "width": 0,
        },
        "shadowOpacity": 0.24,
        "shadowRadius": 4,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      focusable={true}
      hitSlop={
        Object {
          "bottom": 6,
          "left": 6,
          "right": 6,
          "top": 6,
        }
      }
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "margin": 6,
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 18,
              "height": 36,
              "width": 36,
            },
            false,
            undefined,
          ],
        ]
      }
      testID="all-recipes"
    >
      <View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "justifyContent": "center",
              },
              Object {
                "height": 24,
                "width": 24,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "transform": Array [
                  Object {
                    "rotate": "0deg",
                  },
                ],
              }
            }
          >
            <Text
              accessibilityElementsHidden={true}
              allowFontScaling={false}
              importantForAccessibility="no-hide-descendants"
              pointerEvents="none"
              style={
                Array [
                  Object {
                    "color": "#ffffff",
                    "fontSize": 24,
                  },
                  Array [
                    Object {
                      "transform": Array [
                        Object {
                          "scaleX": 1,
                        },
                      ],
                    },
                    Object {
                      "backgroundColor": "transparent",
                    },
                  ],
                  Object {
                    "fontFamily": "Material Design Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              󰺻
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      focusable={true}
      hitSlop={
        Object {
          "bottom": 6,
          "left": 6,
          "right": 6,
          "top": 6,
        }
      }
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "margin": 6,
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 18,
              "height": 36,
              "width": 36,
            },
            false,
            undefined,
          ],
        ]
      }
      testID="my-recipes"
    >
      <View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "justifyContent": "center",
              },
              Object {
                "height": 24,
                "width": 24,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "transform": Array [
                  Object {
                    "rotate": "0deg",
                  },
                ],
              }
            }
          >
            <Text
              accessibilityElementsHidden={true}
              allowFontScaling={false}
              importantForAccessibility="no-hide-descendants"
              pointerEvents="none"
              style={
                Array [
                  Object {
                    "color": "#ffffff",
                    "fontSize": 24,
                  },
                  Array [
                    Object {
                      "transform": Array [
                        Object {
                          "scaleX": 1,
                        },
                      ],
                    },
                    Object {
                      "backgroundColor": "transparent",
                    },
                  ],
                  Object {
                    "fontFamily": "Material Design Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              󰺻
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>,
]
`;
